drivers/adc/ad4170/ad4170.h

```diff
@@ -0,0 +1,1142 @@
+	AD4170_DIG_STR_DEFAULT,
+	/** Increased Drive Strength. */
+	AD4170_DIG_STR_HIGH
+};
+
+/**
+ * @enum ad4170_sdo_rdby_dly
+ * @brief Reset Interface on CS or SCLK.
+ */
+enum ad4170_sdo_rdby_dly {
+	/** Reset on Last SCLK. */
+	AD4170_SDO_RDY_SCLK,
+	/** Reset on /CS Edge. */
+	AD4170_SDO_RDY_CSB
+};
+
+/**
+ * @struct ad4170_pin_muxing
+ * @brief Pin_Muxing register settings.
+ */
+struct ad4170_pin_muxing {
+	/** Enables Current Channel Number Be Output to GPIO Pins. */
+	enum ad4170_chan_to_gpio chan_to_gpio;
+	/** Configures Functionality of DIG_AUX2 Pin. */
+	enum ad4170_dig_aux2_ctrl dig_aux2_ctrl;
+	/** Configures Functionality of DIG_AUX1 Pin. */
+	enum ad4170_dig_aux1_ctrl dig_aux1_ctrl;
+	/** Configures SYNC_IN Pin for ADC Synchronization. */
+	enum ad4170_sync_ctrl sync_ctrl;
+	/** Configures the drive strength of the Digital Outputs. */
+	enum ad4170_dig_out_str dig_out_str;
+	/** Reset Interface on CS or SCLK. */
+	enum ad4170_sdo_rdby_dly sdo_rdby_dly;
+};
+
+/**
+ * @enum ad4170_dclk_div
+ * @brief Continuous Transmit Data Clock Divider.
+ */
+enum ad4170_dclk_div {
+	/** DCLK Equals Master Clock Divide by 1. */
+	AD4170_DCLKDIVBY1,
+	/** DCLK Equals Master Clock Divide by 2. */
+	AD4170_DCLKDIVBY2,
+	/** DCLK Equals Master Clock Divide by 4. */
+	AD4170_DCLKDIVBY4,
+	/** DCLK Equals Master Clock Divide by 8. */
+	AD4170_DCLKDIVBY8
+};
+
+/**
+ * @enum ad4170_mclk_div
+ * @brief Master Clock Divider.
+ */
+enum ad4170_mclk_div {
+	/** Divide by 1. */
+	AD4170_CLKDIVBY1,
+	/** Divide by 2. */
+	AD4170_CLKDIVBY2,
+	/** Divide by 4. */
+	AD4170_CLKDIVBY4,
+	/** Divide by 8. */
+	AD4170_CLKDIVBY8
+};
+
+/**
+ * @enum ad4170_clocksel
+ * @brief ADC Clock Select.
+ */
+enum ad4170_clocksel {
+	/** Internal Oscillator. */
+	AD4170_INTERNAL_OSC,
+	/** Internal Oscillator, Output to XTAL2/CLKIO Pin. */
+	AD4170_INTERNAL_OSC_OUTPUT,
+	/** External Clock Input on XTAL2/CLKIO Pin. */
+	AD4170_EXTERNAL_OSC,
+	/** External Crystal on XTAL1 and XTAL2/CLKIO Pins. */
+	AD4170_EXTERNAL_XTAL
+};
+
+/**
+ * @struct ad4170_clock_ctrl
+ * @brief Clock_Ctrl register settings.
+ */
+struct ad4170_clock_ctrl {
+	/** Continuous Transmit Data Clock Divider. */
+	enum ad4170_dclk_div dclk_divide;
+	/** Master Clock Divider. */
+	enum ad4170_mclk_div clockdiv;
+	/** ADC Clock Select. */
+	enum ad4170_clocksel clocksel;
+};
+
+/**
+ * @enum ad4170_cont_read
+ * @brief Configures continuous Data Register Read/Transmit.
+ */
+enum ad4170_cont_read {
+	/** Disable Continuous Read/Transmit. */
+	AD4170_CONT_READ_OFF,
+	/** Enable Continuous Read. This enables continuous read of the ADC Data register over SPI. */
+	AD4170_CONT_READ_ON,
+	/** Enable Continuous Transmit. This enables continuous transmit of the ADC Data register over TDM. */
+	AD4170_CONT_TRANSMIT_ON
+};
+
+/**
+ * @enum ad4170_mode
+ * @brief ADC Operating Mode.
+ */
+enum ad4170_mode {
+	/** Continuous Conversion Mode. ADC converts continuously on the enabled channel(s) using Sinc-based filters. */
```

Comment:
Should these comments be split on multiple lines?

Reply:
hmmm, astyle didnt complain so i guess it's ok